### PR TITLE
integrating precommit

### PR DESCRIPTION
**Pull request by stahanna**
**Description**

This pull request ...

**Dear reviewer**

Does it solve the issue #5277 ?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-	'numpy>=2.0.0rc1',
-	'setuptools_scm[toml] >= 8',
+    "setuptools >= 65",
+    "Cython",
+    "numpy >= 2.0.0rc1",
+    "setuptools_scm[toml] >= 8",
 ]
 build-backend = "setuptools.build_meta"
 
 [tool.black]
```
Comment (morcuended) [inline pyproject.toml:10]: I guess the black settings will go away if we stick with ruff

Comment (stahanna) [reply pyproject.toml]: ok, thanks for your remark, I removed the [tool.black] part from pyproject.toml

File: pyproject.toml [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-  | \.mypy_cache
-  | \.tox
-  | \.venv
-  | _build
-  | buck-out
-  | build
-  | gammapy/extern
-)/
-'''
+#[tool.black]
```
Comment (registerrier) [inline pyproject.toml:10]: Maybe remove the comments here.

File: examples/tutorials/api/priors.py [hunk 3]
```diff
@@ -71,6 +71,9 @@
     UniformPrior,
 )
 from gammapy.utils.check import check_tutorials_setup
+from gammapy.modeling import PriorParameter
+from gammapy.modeling.models import Model, Prior
+from gammapy.modeling.models import PRIOR_REGISTRY
```
Comment (registerrier) [inline examples/tutorials/api/priors.py:76]: In this specific example I would not change the imports and leave them at their initial place in the the tutorial

File: examples/tutorials/starting/overview.py [hunk 4]
```diff
@@ -75,6 +75,10 @@
 # Check setup
 # -----------
 from gammapy.utils.check import check_tutorials_setup
+from gammapy.maps import Map
+from gammapy.data import EventList
+from gammapy.utils.regions import SphericalCircleSkyRegion
+from gammapy.catalog import SourceCatalog3FHL
```
Comment (registerrier) [inline examples/tutorials/starting/overview.py:81]: In this case too, I would not change the import order.

**Review by registerrier — COMMENTED** [review]
Thanks @stahanna . This is good. There are only few remaining issues:
- remove .ruff.toml as everything is in pyproject.toml
- remove exceution time file
- the import order could be reverted in example/starting/overview.py and examples/api/priors.py

File: examples/tutorials/starting/overview.py (hunk 4 above)
Comment (stahanna) [reply examples/tutorials/starting/overview.py]: it's done, please check if everything is ok now

File: examples/tutorials/analysis-1d/cta_sensitivity.py [hunk 5]
```diff
@@ -49,6 +49,7 @@
 # Check setup
 # -----------
 from gammapy.utils.check import check_tutorials_setup
+from cycler import cycler
```
Comment (morcuended) [inline examples/tutorials/analysis-1d/cta_sensitivity.py:52]: I think cycler import should be moved above with the rest of the imports. Also in other tutorials, the check tutorials setup import has been moved with the rest of the imports.

File: examples/tutorials/api/catalog.py [hunk 6]
```diff
@@ -60,7 +60,7 @@
 # Check setup
 # -----------
 from gammapy.utils.check import check_tutorials_setup
-
+from gammapy.catalog import SourceCatalog4FGL
```
Comment (morcuended) [inline examples/tutorials/api/catalog.py:63]: move import above after CATALOG_REGISTRY import

File: examples/tutorials/api/model_management.py [hunk 7]
```diff
@@ -77,6 +77,7 @@
 # Check setup
 # -----------
 from gammapy.utils.check import check_tutorials_setup
+from gammapy.catalog import SourceCatalog3FHL
```
Comment (morcuended) [inline examples/tutorials/api/model_management.py:80]: same

**Comment by registerrier** [comment]
Ah. This has reverted all the changes I made...

**Comment by registerrier** [comment]
This should be running now. I propose to merge now. OK @QRemy @AtreyeeS ?

**Pull request merged**